localstack/services/apigateway/apigateway_listener.py

```diff
@@ -313,6 +315,60 @@ def return_response(self, method, path, data, headers, response):
             )
 
 
+class RequestValidator:
+    __slots__ = ["context", "apigateway_client"]
+
+    def __init__(self, context: ApiInvocationContext, apigateway_client):
+        self.context = context
+        self.apigateway_client = apigateway_client
+
+    def is_request_valid(self) -> bool:
+        # make all the positive checks first
+        if self.context.resource is None or "resourceMethods" not in self.context.resource:
+            return True
+
+        resource_methods = self.context.resource["resourceMethods"]
+        if self.context.method not in resource_methods:
+            return True
+
+        # check if there is validator for the resource
+        resource = resource_methods[self.context.method]
+        if (
+            "requestValidatorId" not in resource
+            or resource["requestValidatorId"] is None
+            or not resource["requestValidatorId"].strip()
+        ):
+            return True
+
+        # check if there is a validator for this request
+        validator = self.apigateway_client.get_request_validator(
+            restApiId=self.context.api_id, requestValidatorId=resource["requestValidatorId"]
+        )
+        if validator is None:
+            return True
+
+        # are we validating the body?
+        if validator["validateRequestBody"]:
+            # we need a model to validate the body
+            if "requestModels" not in resource or not resource["requestModels"]:
+                return False
+
+            schema_name = resource["requestModels"].get(APPLICATION_JSON)
+            model = self.apigateway_client.get_model(
+                restApiId=self.context.api_id,
+                modelName=schema_name,
+            )
+            if not model:
+                return False
+
+            try:
+                validate(instance=json.loads(self.context.data), schema=json.loads(model["schema"]))
+                return True
+            except ValidationError as e:
+                LOG.error("failed to validate request body", e)
```

Comment:
I'd reduce the log level to `warning` or even `info` here - we're usually using `error` only for fatal internal errors.